File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int f0 = 1, f1 = 1; 
        for (int i = 1; i <= n-1; ++i) {
            int ff = f0; 
            f0 = f1; 
            f1 = ff + f1; 
        }
        return f1; 
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n < 3:
            return n
        one, two = 1, 2
        for _ in range(3, n+1):
            tmp = one + two
            one = two
            two = tmp
        return two
    
        if n <= 2: return n
        one,two=1,2

        for _ in range(3, n+1):
          tmp = one + two
          one = two
          two = tmp
        return two